test/generate-vdm.spec.ts

```diff
@@ -0,0 +1,182 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import GenerateVdm, { FlagsParsed } from '../src/commands/generate-vdm';
+import * as foo from '../src/utils/generate-vdm-util';
+
+const spyToGeneratorSDK = jest.spyOn(foo, 'toGeneratorSDK');
+
+describe('generate-vdm', () => {
+  const pathForTests = path.resolve(__dirname, __filename.replace(/\./g, '-')).replace('-ts', '');
+
+  beforeAll(() => {
+    const pathForResources = path.resolve(__dirname, 'resources', 'template-generator-vdm');
+    fs.copySync(pathForResources, pathForTests);
+  });
+
+  afterAll(() => {
+    fs.removeSync(pathForTests);
+  });
+
+  afterEach(() => {
+    spyToGeneratorSDK.mockClear();
+  });
+
+  beforeEach(() => {
+    spyToGeneratorSDK.mockClear();
+  });
+
+  it('should generate a vdm.', async () => {
+    await GenerateVdm.run(['-i', 'edmxSource', '-o', 'generatedVdm', '--forceOverwrite', '--projectDir', pathForTests]);
+    const files = fs.readdirSync(path.join(pathForTests, 'generatedVdm', 'yy-1-socialnetworkaccount-cds-service'));
+    expect(files.length).toBeGreaterThan(0);
+  }, 60000);
+
+  it('should fail if the mandatory parameters are not there', async () => {
+    try {
+      await GenerateVdm.run([]);
+    } catch (err) {
+      const messageNoFormatting = err.message as string;
+      expect(messageNoFormatting).toContain('-i, --inputDir INPUTDIR');
+    }
+  });
+
+  it('should parse strings and with --no- it should lead to all false.', async () => {
+    const expected = getAllFalse();
+    delete expected.projectDir;
+    try {
+      await GenerateVdm.run([...getInputAllFalse(), '--projectDir', getProjectDir()]);
+    } catch (e) {
+      expect(e.message).toContain('ENOENT: no such file or directory');
+      expect(spyToGeneratorSDK).toHaveReturnedWith(expected);
+    }
+  });
+
+  it('should resolve the projectdir', async () => {
+    const expected = getDefault(getProjectDir());
+
+    try {
+      await GenerateVdm.run(['-i', 'input', '-o', 'output', '--projectDir', getProjectDir()]);
+    } catch (e) {
+      expect(e.message).toContain('ENOENT: no such file or directory');
+      expect(spyToGeneratorSDK).toHaveReturnedWith(expected);
+    }
+  });
+
+  function getProjectDir() {
+    return path.resolve(__dirname, 'generate-vdm-spec');
+  }
+
+  function getInputAndExpected(key: keyof GeneratorOptionsSDK): { expected: GeneratorOptionsSDK; args: string[] } | undefined {
+    const args = ['-i', 'input', '-o', 'output', '--projectDir', getProjectDir()];
+    const expected = getDefault(getProjectDir());
+
+    const option = generatorOptionsSDK[key];
+    if (option && option.type === 'boolean') {
+      const casted = key as keyof GeneratorOptionsSDK;
+      if (!option.default) {
+        args.push(`--${key}`);
+        (expected[casted] as any) = true;
+      } else {
+        args.push(`--no-${key}`);
+        (expected[casted] as any) = false;
+      }
+      return { args, expected };
+    }
+    return undefined;
```

Comment:
This can be removed. No return is equivalent to `return undefined`

Reply:
learned something.

Reply:
learned something.